strawberryfields/ops.py

```diff
@@ -1435,16 +1389,17 @@ def __init__(self, s=1):
 
     def _decompose(self, reg, **kwargs):
         s = self.p[0]
-        r = arcsinh(-s/2)
-        theta = 0.5*arctan2(-1.0/cosh(r), -tanh(r))
-
-        BS1 = BSgate(theta, 0)
-        BS2 = BSgate(theta+pi/2, 0)
+        r = pf.asinh(-s/2)
+        #theta = 0.5*pf.atan2(-1.0/pf.cosh(r), -pf.tanh(r))
+        # FIXME in sympy 1.4 atan2._eval_evalf() has a bug, it does not work with Symbol._eval_evalf().
+        # This is a partial workaround for s>0. When sympy is fixed (in version 1.5?), go back to using pf.atan2.
+        # If s<0 we would need to add pi/2 to theta, but since s is symbolic at this point, it'd be complicated.
```

Comment:
So what is our course of action in the case that s<0?

Reply:
Hmm, maybe add `sympy.Heaviside(-s)*np.pi/2` to theta? But Heaviside is undefined at zero, which could also cause some problems.

Another option would be to evaluate the parameters right here until sympy is fixed (should happen when the next version is released this fall), and leave a note to warn the user of the unexpected behavior.